### ISSUE TITLE
Multiple webhooks failing to be notified

### ISSUE THREAD

**Issue by rob-baldwin** [Xabaril/AspNetCore.Diagnostics.HealthChecks#101]
I believe there's a small issue with the webhook notifications when there are multiple webhooks setup.  The issue is related to the `MinimumSecondsBetweenFailureNotifications` setting, which correctly prevents multiple notifications from being triggered within a time period.  However the way in which multiple webhooks are being handled means that only the first webhook will ever be notified.

The code currently looks like this (from WebHookFailureNotifier.cs):

```
foreach (var webHook in _settings.Webhooks)
{
    var payload = isHealthy ? webHook.RestoredPayload : webHook.Payload;
    payload = payload.Replace(Keys.LIVENESS_BOOKMARK, name);

    if (!await IsNotifiedOnWindowTime(name, isHealthy))
    {
        payload = payload.Replace(Keys.FAILURE_BOOKMARK, failure);

        await SaveNotification(new HealthCheckFailureNotification()
        {
            LastNotified = DateTime.UtcNow,
            HealthCheckName = name,
            IsUpAndRunning = isHealthy
        });

        await SendRequest(webHook.Uri, webHook.Name, payload);
    }
    else
    {
        _logger.LogInformation("Notification is sent on same window time.");
    }
}
```

I would propose changing the order of the logic to something like:

```
if (!await IsNotifiedOnWindowTime(name, isHealthy))
{
    await SaveNotification(new HealthCheckFailureNotification()
    {
        LastNotified = DateTime.UtcNow,
        HealthCheckName = name,
        IsUpAndRunning = isHealthy
    });

    foreach (var webHook in _settings.Webhooks)
    {
        var payload = isHealthy ? webHook.RestoredPayload : webHook.Payload;
        payload = payload.Replace(Keys.LIVENESS_BOOKMARK, name);
        payload = payload.Replace(Keys.FAILURE_BOOKMARK, failure);

        await SendRequest(webHook.Uri, webHook.Name, payload);
    }
}
else
{
    _logger.LogInformation("Notification is sent on same window time.");
}
```

I'm happy to raise a PR for this.

**Comment by unaizorrilla** [comment]
A PR is welcome!

**Comment by CarlosLanderas** [comment]
@rob-baldwin you are completely right, they IsNotifiedOnWindowTime should be outside the foreach. Thanks for reporting, if you wan't to create a PR with the change we will merge it fast as is evident that the logic is wrong.

**Comment by unaizorrilla** [comment]
Fixed on #104

**Comment by rob-baldwin** [comment]
Wonderful, thank you for the prompt responses @unaizorrilla & @CarlosLanderas
